### ISSUE TITLE
Wordpress credentials - Localhost HTTPS server self signed

### ISSUE THREAD

**Issue by pablorq** [n8n-io/n8n#8151]
**Describe the bug**
The configuration of credentials for Wordpress in localhost with self signed certificates, doesn't work.

**To Reproduce**

1. Go to 'Credentials'
2. Add a new credential for Wordpress.
3. Configure the user an application password.
4. For the server, enter the url of a self signed certificates wordpress server
5. See error.

**Expected behavior**
Connection and validation of the credentials.

**Environment (please complete the following information):**
- OS: Ubuntu Linux 23.10
- n8n Version: 1.20.0
- Node.js Version: ?
- Database system: postgresql
- Docker container
- Operation mode [e.g. own; operation modes are `own`, `main` and `queue`. Default is `main`]

**Additional context**
The application password was previously configured on the Wordpress site.

This is the screenshot of the error:
![Screenshot 2023-12-26 at 17-30-40 n8n - Workflow Automation](https://github.com/n8n-io/n8n/assets/774839/93053a1d-7f50-4c87-ab91-e4722ac66b62)

**Comment by pablorq** [comment]
**Testing**
The 1st test was to check if the connection could be tested from inside the container, to check connectivity an more. After install `curl`, this test shows:


```
$ curl -X POST -i \
-H "Content-Type: application/x-www-form-urlencoded" \
-d "client_id=myuser&client_secret=mypassword&grant_type=application_password" \
https://han.local:8120

curl: (60) SSL certificate problem: unable to get local issuer certificate
More details here: https://curl.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.
```

This shows an problem with the certificate issuer, which is normal in a self signed certificate. Trying the insecure way, it works:

```
$ curl -k -X POST -i \
-H "Content-Type: application/x-www-form-urlencoded" \
-d "client_id=myuser&client_secret=mypassword&grant_type=application_password" \
https://han.local:8120
```

This means that the connectivity is good, but there's a problem with the certificate issuer.

For that, I've installed the certificate issuer in the container, copying the `.pem` file to  `/usr/local/share/ca-certificates/` and running `update-ca-certificates`. After this steps, the same 1st command that failed, now works well:

```
$ curl -X POST -i \
-H "Content-Type: application/x-www-form-urlencoded" \
-d "client_id=myuser&client_secret=mypassword&grant_type=application_password" \
https://han.local:8120
```

Testing again in n8n the credentials for wordpress, it still fails.

Then I created a simple workflow with a HTTP node and imported the configuration from curl, but it also fails with the same error:

![Screenshot 2023-12-26 at 17-48-54 n8n - ⚠️ Test - Post to Wordpress](https://github.com/n8n-io/n8n/assets/774839/1801d721-e395-4c24-bb9a-c1555a02e710)

Any idea how to solve this?

**Comment by mbrother2** [comment]
Hi,

Did you try "Ignore SSL Issues" option in HTTP Request node? Related issue: https://community.n8n.io/t/self-signed-certificate/16067

**Comment by pablorq** [comment]
Hi @mbrother2 !

Sorry if I didn't express myself correctly. The problem is to setting the credentials for a wordpress site in server (could be localhost) with a self signed certificate.

The HTTP Request is just a test. I know that I can set "Ignore SSL Issues", but in that case I was testing the installation of the CA in the n8n container to check if that could be a workaround to the setting credentials issue. If I set "Ignore SSL Issues", I can't see if the HTTP Request is using the local CA to authorize the self signed certificate.

The idea is the check why the wordpress credential settings doesn't work with a self signed certificate and try to see how to improve it.

**Comment by Joffcom** [comment]
Hey @pablorq,

Thanks for raising this issue, I took a look and this isn't really a bug and it is more of a feature request which is better suited to our support forum.

I have put in a PR to add this to a future release, For now I am going to mark this as closed though. Once we have merged and released a version of n8n that contains this change I will let you know.

**Comment by Joffcom** [comment]
Fix got released with [`n8n@1.23.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.23.0)
